specs/RasterizationScale.md

```diff
@@ -0,0 +1,164 @@
+When the host app has a DPI awareness of PerMonitorV2, then all child HWNDs receive WM_DPICHANGED.
+The WebView can check the current monitor DPI scale when it receives WM_DPICHANGED.
+But if the host app is just PerMonitor aware (not v2), then only the top level window receives WM_DPICHANGED.
+The WebView could try to use other metrics (like resize) to check the DPI, but resize is not guaranteed in that scenario.
+From [GitHub feedback](https://github.com/MicrosoftEdge/WebView2Feedback/issues/65),
+we also know that there are consumers of WebView that want to explicitly set a custom DPI scale for WebView.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+
+# Description
+To allow the app to control the DPI scale of WebView, we are adding a `RasterizationScale` property to the CoreWebView2Controller interface.
+The RasterizationScale property controls the DPI scaling for all UI in WebView.
+
+In order to maintain compatibility with apps developed before this API existed,
+WebView2 continues to detect monitor scale changes by default and will update the RasterizationScale property.
+When the RasterizationScale property is updated, the `RasterizationScaleChanged` event is raised.
+The app can tell WebView2 to stop updating the RasterizationScale property by changing
+`ShouldDetectMonitorDpiScaleChanges` from the default value of true, to false.
+
+# Examples
+The following code snippets show how to use the RasterizationScale and ShouldDetectMonitorScaleChanges properties,
+and now to listen to the RasterizationScaleChanged event.
+## Win32 C++
+```cpp
+void ViewComponent::SetRasterizationScale(float additionalScale)
+{
+    CHECK_FAILURE(m_controller->put_ShouldDetectMonitorScaleChanges(FALSE));
+    m_webviewAdditionalRasterizationScale = additionalScale;
+    // RasterizationScale is typically the monitor DPI scale and text scaling, but
+    // the app could add an additional scale for its own scenarios.
+    double rasterizationScale =
+        additionalScale * m_appWindow->GetDpiScale() * m_appWindow->GetTextScale();
+    CHECK_FAILURE(m_controller->put_RasterizationScale(rasterizationScale));
+}
+
+Callback<ICoreWebView2RasterizationScaleChangedEventHandler>(
+    [this](ICoreWebView2Controller* sender, IUnknown* args) -> HRESULT {
+        double rasterizationScale;
+        CHECK_FAILURE(m_controller->get_RasterizationScale(&rasterizationScale));
+
+        std::wstring message = L"WebView2APISample (RasterizationScale: " +
+            std::to_wstring(int(rasterizationScale * 100)) + L"%)";
+        SetWindowText(m_appWindow->GetMainWindow(), message.c_str());
+        return S_OK;
+    })
+.Get(), &m_rasterizationScaleChangedToken));
+```
+## .Net
+```c#
+public void SetRasterizationScale(double additionalScale)
+{
+    CoreWebView2Controller.ShouldDetectMonitorScaleChanges = false;
+    m_webviewAdditionalRasterizationScale = additionalScale;
```

Comment:
Either this sample isn't quite realistic, or I'm not understanding it. I would expect a method to set additionalScale, and then a separate method to recalc and set RasterizationScale. That's because the recalc/set needs to also be called any time the window scaling changes (for example if you drag the window to a different monitor).

Reply:
This was pulled from our sample app, which uses this method for both. When needed for recalc, m_webviewAditionalRasterizationScale is passed in, so the value is reset to the same value, and everything else is recalculated.